packages/components/fields/select-field/src/select-field.tsx

```diff
@@ -0,0 +1,350 @@
+   */
+  backspaceRemovesValue?: boolean;
+  /**
+   * Map of components to overwrite the default ones, see what components you can override
+   */
+  components?: {
+    [key: string]: () => void;
+  };
+  /**
+   * Custom method to filter whether an option should be displayed in the menu
+   */
+  filterOption?: ReactSelectProps['filterOption'];
+  /**
+   * The id to set on the SelectContainer component
+   */
+  containerId?: string;
+  /**
+   * Is the select value clearable
+   */
+  isClearable?: boolean;
+  /**
+   * Is the select disabled
+   */
+  isDisabled?: boolean;
+  /**
+   * Is the select read-only
+   */
+  isReadOnly?: boolean;
+  /**
+   * Override the built-in logic to detect whether an option is disabled
+   */
+  isOptionDisabled?: ReactSelectProps['isOptionDisabled'];
+  /**
+   * Support multiple selected options
+   */
+  isMulti?: boolean;
+  /**
+   * Whether to enable search functionality
+   */
+  isSearchable?: boolean;
+  /**
+   * Maximum height of the menu before scrolling
+   */
+  maxMenuHeight?: number;
+  /**
+   * Dom element to portal the select menu to
+   */
+  menuPortalTarget?: ReactSelectProps['menuPortalTarget'];
+  /**
+   * z-index value for the menu portal
+   */
+  menuPortalZIndex?: number;
+  /**
+   * whether the menu should block scroll while open
+   */
+  menuShouldBlockScroll?: boolean;
+  /**
+   * Name of the HTML Input (optional - without this, no input will be rendered)
+   */
+  name?: string;
+  /**
+   * Can be used to render a custom value when there are no options (either because of no search results, or all options have been used, or there were none in the first place). Gets called with { inputValue: String }.
+   * <br />
+   * `inputValue` will be an empty string when no search text is present.
+   */
+  noOptionsMessage?: ReactSelectProps['noOptionsMessage'];
+  /**
+   * Handle blur events on the control
+   * <br />
+   * Signature: `(event) => void`
+   */
+  onBlur?: () => void;
+  /**
+   * Called with a fake event when value changes. The event's target.name will be the name supplied in props. The event's target.value will hold the value.
+   * <br/>
+   * The value will be the selected option, or an array of options in case isMulti is true.
+   * <br />
+   * Signature: `(event) => void`
+   */
+  onChange?: () => void;
+  /**
+   * Handle focus events on the control
+   * <br />
+   * Signature: `(event) => void`
+   */
+  onFocus?: () => void;
+  /**
+   * Handle change events on the input
+   * <br />
+   * Signature: `(newValue, actionMeta) => void`
+   */
+  onInputChange?: () => void;
+  /**
+   * Array of options that populate the select menu
+   */
+  // options?: OptionsOrGroups<unknown, GroupBase<unknown>> | boolean;
+  options?: TOptions;
+  showOptionGroupDivider?: boolean;
+  /**
+   * Placeholder text for the select value
+   */
+  placeholder?: string;
+  /**
+   * Sets the tabIndex attribute on the input
```

Comment:
For `react-select` props maybe we could include a link to its props docs as we've done in other or our components ([example](https://github.com/commercetools/ui-kit/blob/main/packages/components/inputs/async-select-input/src/async-select-input.tsx#L92))